course7/homeworks/anton/ls/main.cc

```diff
@@ -0,0 +1,135 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+struct configuration
+{
+    std::string file_path = ".";
+    bool show_help = false;
+    bool show_tree = false;
+    bool show_long = false;
+};
+
+configuration parse_args(int argc, char *argv[]);
+void print_file_tree(const fs::path& dir, const configuration& config, const std::string& filler = "");
+size_t get_file_size(const fs::path&);
+
+int main(int argc, char *argv[])
+{
+    try
+    {
+        const configuration& config = parse_args(argc, argv);
```

Comment:
I don't see much point in using const ref here.
Returning stuff is cheap; an instance must exist anyway.